extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/options/TlsCertificateReloadUtils.java

```diff
@@ -0,0 +1,133 @@
+package io.quarkus.vertx.http.runtime.options;
+
+import static io.quarkus.vertx.http.runtime.options.HttpServerOptionsUtils.getFileContent;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.Callable;
+import java.util.function.Function;
+
+import org.jboss.logging.Logger;
+
+import io.quarkus.vertx.http.runtime.HttpConfiguration;
+import io.vertx.core.AsyncResult;
+import io.vertx.core.Future;
+import io.vertx.core.Handler;
+import io.vertx.core.Vertx;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.core.http.HttpServer;
+import io.vertx.core.http.HttpServerOptions;
+import io.vertx.core.net.KeyStoreOptions;
+import io.vertx.core.net.PemKeyCertOptions;
+import io.vertx.core.net.SSLOptions;
+
+/**
+ * Utility class to handle TLS certificate reloading.
+ */
+public class TlsCertificateReloadUtils {
+
+    public static long handleCertificateReloading(Vertx vertx, HttpServer server,
+            HttpServerOptions options, HttpConfiguration configuration) {
+        // Validation
+        if (configuration.ssl.certificate.reloadPeriod.isEmpty()) {
+            return -1;
+        }
+        if (configuration.ssl.certificate.reloadPeriod.get().toMillis() < 1000) {
```

Comment:
Maybe we should be even stricter?

Reply:
what do you have in mind?

Yes, reloading often is counter-productive.

Reply:
I'm thinking something like 30  seconds.

Reply:
Deal!